src/frontend/query_commands.ml

```diff
@@ -827,7 +829,7 @@ let dispatch pipeline (type a) : a Query_protocol.t -> a =
 
   | Path_list `Build ->
     let config = Mpipeline.final_config pipeline in
-    Mconfig.(config.merlin.build_path.visible @ config.merlin.build_path.hidden)
+    Mconfig.(config.merlin.build_path)
```

Comment:
I don't really know what this is used for, but why shouldn't it include the `-H` paths?

Reply:
I don't know what this is used for either, I think I just overlooked this. I think you're probably right that it should include hidden deps.